### ISSUE TITLE
IllegalStateException after obfuscation

### ISSUE THREAD

**Issue by tarasmorskyi-mediamonks** [realm/realm-kotlin#778]
Receiving ISE on creation of RealmConfiguration on release build which got obfuscated with Dexguard.
Error message:
```
Caused by: java.lang.IllegalStateException: Couldn't find companion object of class 'ProductEntity'.
2022-04-06 10:55:01.901 32219-32219/? E/AndroidRuntime: A common cause for this is when the `io.realm.kotlin` is not applied to the Gradle module that contains the 'ProductEntity' class.
```
Added these rules for RealmObject classes which didn't help:
```
-keep,includedescriptorclasses class com.projectname.data.vo.**$$serializer { *; }
-keepclassmembers class com.projectname.data.vo.** { *** Companion; }
-keepclasseswithmembers class com.projectname.data.vo.** {
    kotlinx.serialization.KSerializer serializer(...);
}
```
Is there any additional configuration needed for it? Or is it rather general misconfiguration of Dexguard on my side? Any kind of these issues are not happening on pre-KMM version of project.

**Comment by nhachicha** [comment]
Hi @tarasmorskyi-mediamonks have you tried the Proguard conf specified here https://github.com/realm/realm-kotlin-samples/blob/main/Bookshelf/androidApp/proguard-rules.pro ?

**Comment by tarasmorskyi-mediamonks** [comment]
Hi @nhachicha, how are these rules different to what I've shared above? There are only first few lines extra which are Kotlin Serialisation specific which I had already but didn't mention here as RealmObjects are not annotated as Serialisable

**Comment by nhachicha** [comment]
@tarasmorskyi-mediamonks can you please share more on your project to try to reproduce it:
- Version of Kotlin 
- Version of Realm Kotlin
- Android or KMM project
- Model classes
- Gradle configurations

**Comment by tarasmorskyi-mediamonks** [comment]
Kotlin 1.6.10
Realm Kotlin 0.10.2
It's a company project, so I cannot share access to repo
It happens to any model, even as simple as this one:
```
open class SearchedProduct : RealmObject {

    @PrimaryKey
    var id: Long = SearchedProductId

    var productId: RealmList<String> = realmListOf()
    var updateTime: Long = 0
}
```
What part of Gradle configuration are you interested in?

**Comment by nhachicha** [comment]
- Does this also fail when using only Proguard (instead of Dexguard)?
- Is the `ProductEntity` class within the same package as `SearchedProduct`  (i.e `com.projectname.data.vo`)?
- Is this a KMM project or Android only project?

**Comment by tarasmorskyi-mediamonks** [comment]
- Also fails with Proguard
- yes, I have all packages with models covered with rules mentioned above
- It's in the multiplatform module, but it's used only by Android

**Comment by nhachicha** [comment]
@tarasmorskyi-mediamonks hmm https://github.com/realm/realm-kotlin-samples/tree/main/Bookshelf checks all the boxes (uses ProGaurd, model in multiplatform module and same Proguard rules ...) I wonder if you can manage to reproduce the issue using the Bookshelf as a starting base?

**Comment by tarasmorskyi-mediamonks** [comment]
In the end issue was not with Proguard/Dexguard. It got fixed after updating gradle plugin from 7.0.4 to 7.1.2. Cannot understand why it didn't affect the debug build though.
